base/logging.jl

```diff
@@ -359,29 +420,21 @@ function default_group_code(file)
 end
 
 
-# Report an error in log message creation (or in the logger itself).
+# Report an error in log message creation
 @noinline function logging_error(logger, level, _module, group, id,
-                                 filepath, line, @nospecialize(err))
+                                 filepath, line, @nospecialize(err), real::Bool)
     if !_invoked_catch_exceptions(logger)
-        rethrow(err)
-    end
-    try
-        msg = "Exception while generating log record in module $_module at $filepath:$line"
-        handle_message(
-            logger, Error, msg, _module, :logevent_error, id, filepath, line;
-            exception=(err,catch_backtrace())
-        )
-    catch err2
-        try
-            # Give up and write to stderr, in three independent calls to
-            # increase the odds of it getting through.
-            print(stderr, "Exception handling log message: ")
-            println(stderr, err)
-            println(stderr, "  module=$_module  file=$filepath  line=$line")
-            println(stderr, "  Second exception: ", err2)
-        catch
-        end
+        real ? rethrow(err) : throw(err)
     end
+    msg = try
+              "Exception while generating log record in module $_module at $filepath:$line"
+          catch ex
+              "Exception handling log message: $ex"
```

Comment:
No, we can't just keep going deep down the error chain. Eventually you have to admit you've broken everything.

Reply:
The point is to fall back to emitting a message which can't depend on user-defined code.

We can clearly do this.  I think you're asserting that you don't want to?

Reply:
Your "correction" still depends on user-defined code being valid, so, yes, clearly we can't do that. The subsequent lines of code need to run yet more user-defined code (and more complex code) so there's little point in adding try/catch here while forgetting to handle most of the actual sources of errors.

Reply:
Seems we're talking past each other here in some way, because I don't understand this statement.

Let me attempt to spell out my reasoning very explicitly.

The following code formats potentially user defined types in both branches, and these arise from the logging macro

```juila
msg = try
    # The user may have passed some odd type for _module or filepath or line here
    # so this may fail.
    "Exception while generating log record in module $_module at $filepath:$line"
catch ex
    # `ex` here is the potentially-user-defined exception resulting from running user-defined `string` functions,
    # so may fail.
    # Granted, this is unlikely
    "Exception handling log message: $ex"
end
```

The following invocation of user-defined code is related to the installed logger (which may internally protect itself with a try-catch of its own, should it so choose)

```julia
    handle_message(
        logger, Error, msg, _module, :logevent_error, id, filepath, line;
        exception=err)
```

So to me it seems there's a clear separation of the user code which
* emits the log record, vs
* consumes the log record

and the former is what I'm arguing should be wrapped in various stupidly-nested try-catch statements.

Reply:
"protect" is just a relatively useless concept here, so I'm intentionally not attempting to pretend to do it excessively

Reply:
Well I don't understand that comment; saying it's a "useless concept" just leaves me guessing here which is a bit disappointing. I want to know the *why* here, not *what*.

But anyway, the actual technicality here is not a hill I'm willing to die on.